packages/ember-glimmer/tests/integration/components/curly-components-test.js

```diff
@@ -2101,4 +2101,40 @@ moduleFor('Components test: curly components', class extends RenderingTest {
 
     this.assertText('initial value');
   }
+
+  ['@htmlbars ensure attributes which have been passed in can be mutated, when they are mutated the external source is updated (#11192)']() {
+    let component;
+    let FooBarComponent = Component.extend({
+      init() {
+        this._super(...arguments);
+        component = this;
+        this.set('bar', 'hellllloooooo');
+      }
+    });
+
+    this.registerComponent('foo-bar', {
+      ComponentClass: FooBarComponent,
+      template: ''
+    });
+
+    expectDeprecation(() => {
+      this.render('{{localBar}}{{foo-bar bar=localBar}}', {
+        localBar: undefined
+      });
+    }, /You modified localBar twice in a single render/);
+
+    this.assertText('hellllloooooo');
```

Comment:
I would not have expected this to work, since I would not expect that set() to cause a change event that would flow back to previous sibling view, by what magic does this work? and people really do this, I hope not...